# HG changeset patch
# User Cole Robinson <crobinso@redhat.com>
# Date 1466514092 14400
#      Tue Jun 21 09:01:32 2016 -0400
# Node ID 8ee2439ccb6120cee4781630a632367ca31d7d97
# Parent  389fb0b3e120b226186e5a61214dd7d3dff3959b
viewers: spice: Catch failure to setup usbdev manager

Since some distros like openbsd don't compile support for
usb redirection, which makes this fail

https://bugzilla.redhat.com/show_bug.cgi?id=1348479

diff --git virtManager/viewers.py virtManager/viewers.py
--- virtManager/viewers.py
+++ virtManager/viewers.py
@@ -538,16 +538,23 @@ class SpiceViewer(Viewer):
         GObject.GObject.connect(self._spice_session, "channel-new",
                                 self._channel_new_cb)
 
-        self._usbdev_manager = SpiceClientGLib.UsbDeviceManager.get(
-                                    self._spice_session)
-        self._usbdev_manager.connect("auto-connect-failed",
-                                    self._usbdev_redirect_error)
-        self._usbdev_manager.connect("device-error",
-                                    self._usbdev_redirect_error)
+        # Distros might have usb redirection compiled out, like OpenBSD
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1348479
+        try:
+            self._usbdev_manager = SpiceClientGLib.UsbDeviceManager.get(
+                                        self._spice_session)
+            self._usbdev_manager.connect("auto-connect-failed",
+                                        self._usbdev_redirect_error)
+            self._usbdev_manager.connect("device-error",
+                                        self._usbdev_redirect_error)
 
-        autoredir = self.config.get_auto_redirection()
-        if autoredir:
-            gtk_session.set_property("auto-usbredir", True)
+            autoredir = self.config.get_auto_redirection()
+            if autoredir:
+                gtk_session.set_property("auto-usbredir", True)
+        except:
+            self._usbdev_manager = None
+            logging.debug("Error initializing spice usb device manager",
+                exc_info=True)
 
 
     #####################
